app/bundles/LeadBundle/Model/Service/DeviceCreatorService/DeviceCreatorService.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * @copyright   2016 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Model\Service\DeviceCreatorService;
+
+use DeviceDetector\DeviceDetector;
+use Mautic\LeadBundle\Entity\Lead;
+use Mautic\LeadBundle\Entity\LeadDevice;
+
+/**
+ * Class DeviceCreatorService.
+ */
+final class DeviceCreatorService implements DeviceCreatorServiceInterface
+{
+    /**
+     * @param DeviceDetector $deviceDetector
+     * @param Lead           $assignedLead
+     *
+     * @return LeadDevice|null Null is returned if device can't be detected
```

Comment:
I don't see any possible way how this method could return null.

Reply:
Since this method only implements interface, it's interface that states it could be null. It's just keeping back doors open for case of problems in detecting in future. I could do that or throw exception, maybe exception would be better